git_hub_contributors/contributors.js

```diff
@@ -0,0 +1,61 @@
+const gitHubUrl = 'https://api.github.com/';
+const otherParams = {
+    headers: {
+        "content-type": "application/json; charset=UTF-8",
+        "Accept": "application/vnd.github.v3+json"
+    },
+    method: "GET"
+};
+
+function getContributors(repo) {
+    const repoUrl = gitHubUrl + 'orgs/' + repo + '/repos';
+    console.log('repoURL:', repoUrl);
+    const fetchedRepos = fetch(repoUrl, otherParams)
+        .then((response) => {
+            return response.json();
+        })
+        .then((result) => {
+            // console.log("first result:", result);
+            return result;
+        })
+        .then((repos) => {
+            for (let i = 0; i < 3; i++) {
+                // console.log(repos[i]);
+                const {contributors_url} = repos[i];
+                const {name} = repos[i];
+                // console.log(contributors_url);
+                contributor_info(contributors_url, name);
+            }
+        })
+        .catch((error) => {
+            console.log("BAD!", error);
```

Comment:
in most cases, consumer of a node.js module is not a human, but another module,
how it can know that something went `BAD!`

Reply:
what should I return here?

Reply:
the function `getContributors` must return promise for result, the promise must be resolved with a result or rejected with an error. So you can remove the `.catch` at all in this case. The `.catch` may be useful if caught error is handled somehow.

you should return the entire chain from the function
```
return fetch(/*...*/)
.then(/*...*/)
```

the `const repoContribMap = new Map();` should be created inside the `getContributors` and should to be resolved when it is ready. Also, I'm not sure it should be a `Map` — there is an object described as a result of the task.

all handling of a result or an error (like `console.log`) should be performed outside the module, like this.
```
getContributors('someOrganization')
.then(console.log.bind(console))
.catch(console.error.bind(console))
```